### DISCUSSION TITLE
Duda

### DISCUSSION THREAD

**Discussion by juanjoc711** [mmasias/pyReposU#3] [General]
Buenas @mmasias estaba trabajando en el proyecto y me surgió una duda que bueno podemos dejar para más adelante  pero por si tú ya tienes claro como lo quieres.

La duda es si la idea sería solo tener un back y un front que se actualiza en tiempo real o si también habrá una base de datos detrás que almacene toda la información.

No es algo tan importante de momento pero es de cara a futuras tareas relacionadas con eso que se me van ocurriendo y se podrían así ir dejando abiertas para en un futuro tenerlas en cuenta y desarrollarlas.

**Comment by mmasias** [comment]
El front me preocupa menos porque con un buen back, la información estructurada y un api podemos consumirlo via script (o completar las info si hace falta)

**Reply by juanjoc711** [reply]
Sii, lose de hecho el front yo lo dejaría con un excel al menos hasta que ya haya bastantes funcionalidades y un buen back, pero mi duda era más si la idea sería también hacer una base de datos por detrás.

Basicamente porque en lo que estoy viendo buscando sobre otras universidades que usan github y demás es que uno de los mayores problemas que ven muchos profesores es que aunque usen github esta no les sirve legalmente como manera de guardar tanto datos de los alumnos como sus tareas. No sé si esto es realmente problema para nosotros, pero por si sí, se podría crear una base de datos detrás aunque sea centrada solo en guardar de forma segura las tareas de los alumnos para tener registros de ellas al menos el tiempo que legalmente se solicita a la universidad guardarlas por si en github hay algún problema.

**Comment by mmasias** [comment]
Ostras, eso sí es interesante. ¿Puedes poner los enlaces dónde has visto (o
estás viendo) eso?

Formalmente queda registro de todo, o al menos es lo que tengo entendido y
comprobado.

**Reply by juanjoc711** [reply]
Claro, te lo comparto. Encontré muchas cosas aún no tuve tiempo de leerme en profundidad la mayoría, pero me he centrado en un trabajo de fin de master de Joseph Feliciano un alumno de la Universidad de Victoria.

En su trabajo, se centrabásicamente en valuar la viabilidad de GitHub en la educación, Identificar beneficios y retos para los instructores y analizar la perspectiva de los estudiantes.

Lo interesante es que para su trabajo, entrevistó  a muchos profesores y alumnos que usaron github en las clases, tanto algunos solo para subir material como otros también para las tareas trabajos en equipos etc.

Y entre otras muchas cosas que iré redactando ahora que ya lo leí prácticamente entero, una de las cosas que me dejó pensando fue que algunos profesores comentan que algunas universidades o regulaciones estadounidenses prohíben almacenar datos estudiantiles en servidores externos a la institución, sobre todo datos como sus puntuaciones y demás. Si queda todo registrado de forma segura no nos será problema ya que las calificaciones por nuestra parte serán privadas ya sea en un excel o la forma de visualización que decidamos, pero bueno igualmente el trabajo tiene muchas ideas interesantes que podrían servirnos, te paso algunas de ellas enfocadas a nuestro desarrollo:
- Ya que sería un sistema bastante diferente para quien este acostumbrado a aulas virtuales habituales, crear un buen tutorial tanto para profesores como alumnos.
- Tener en cuanta que algunos formatos habituales en entregas github no los procesa para poder verlos desde su propia web (mencionan pdf y excel, aunque lo he comprobado y pdf a día de hoy ya sí que lo procesa sin tener que descargarlo, pero se podría pensar en alguna solución para excel u otros posibles).
- Otra idea que saco es que lo más importante es preparar una buena herramienta para la parte de evaluación ya que es precisamente de lo que carece github, pero bueno es precisamente lo que pretendemos hacer.
- Algunos mencionan las notificaciones de github como un problema, ya que o recibes todas, o solo las que lleven tu @ o ninguna, sobre todo estudiantes se quejan de esto. Podríamos crear nuestras propias notificaciones en el framework si hicieramos también la parte de estudiantes.
- A algunos estudiantes les preocupa el usar repos públicos de github ya que precisamente al ser públicos si trabajaron mal en alguna tarea o asignatura eso podría perjudicarles a futuro.
- Otros estudiantes lo ven útil sobre todo en asignaturas con bastante libertad, de trabajos a largo plazo o en equipo y similares, no tanto para todas las tareas.
- Una de las opciones que dael autor delproyecto para solucionar la privacidad es gitlab ya que al parecer tiene mas opciones de privacidad (no sé si esto es así aún no lo pude investigar, en las prácticas usaba gitlab en CIC, pero yo creo que github tiene herramientas suficientes de privacidad y visualmente en mi opinión es mejor que gitlab).


Te dejo el trabajo por si quieres ojearlo, aunque a medida que vaya terminando de leer todo lo que he recopilado y lo que siga recopilando iré añadiéndolo en la carpeta estado del arte que creé esta mañana. He abierto un nuevo issue donde iré escribiendo un mensaje cada vez que termine de añadir a esa carpeta algo de un artículo y te etiquetaré para que puedas ir viendo el avance de cada uno.

http://chisel.cs.uvic.ca/theses/Feliciano_Joseph_MSc_2015.pdf
